src/java/com/puppetlabs/puppetserver/pool/JRubyPool.java

```diff
@@ -307,6 +307,19 @@ public void lock() throws InterruptedException {
     }
 
     @Override
+    public boolean isLocked() {
+        boolean locked;
+        final ReentrantLock lock = this.queueLock;
+        lock.lock();
+        try {
```

Comment:
A try/finally here may not strictly be necessary since `.isPoolLockHeld` just does a variable comparison.  Thought it would be safest to follow the pattern in other methods and future-proof this in the event that `.isPoolLockHeld` might ever have to do something more involved that could throw an exception.